deepmd/utils/data_system.py

```diff
@@ -84,6 +84,12 @@ def __init__ (self,
                     modifier = modifier, 
                     trn_all_set = trn_all_set
                 ))
+        if self.data_systems[0].mixed_type:
+            for data_sys in self.data_systems[1:]:
+                assert data_sys.mixed_type, "all systems must have the same format for mixed_type!"
```

Comment:
The error message is not clear... 
"if one of the system is of mixed_type, then all of the systems should be of mixed_type"

But, the check is only triggered when the data_system[0] is mixed_type. Suppose the case data_system[0] is normal system and data_system[1] is mixed_type, the current check cannot catch the case. 

We should raise error if **ANY** of the data_system is mixed_type and at least one of them is not.